code/modules/projectiles/guns/ballistic/minigun.dm

```diff
@@ -172,3 +174,25 @@
 /obj/item/gun/ballistic/minigunosprey/dropped(mob/living/user)
 	. = ..()
 	ammo_pack.attach_gun(user)
+
+/obj/item/gun/ballistic/minigunosprey/ui_action_click(mob/user, actiontype)
+	if(istype(actiontype, /datum/action/item_action/toggle_firemode))
+		burst_select()
+	else
+		..()
+
+/obj/item/gun/ballistic/minigunosprey/proc/burst_select()
+	var/mob/living/carbon/human/user = usr
```

Comment:
```suggestion
	var/mob/living/carbon/user = usr
```

Reply:
also why is this defined in the first place it doesn't change anything cause you'll need to have the action to activate it anyways